### ISSUE TITLE
Playing multiple AudioStreams together causes alterations of pitch or volume

### ISSUE THREAD

**Issue by Kadmioo** [godotengine/godot#96249]
### Tested versions

- Reproducible in v4.3.stable.mono.official [77dcf97d8]

### System information

Godot v4.3.stable.mono - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2080 Super with Max-Q Design (NVIDIA; 32.0.15.6081) - Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz (16 Threads)

### Issue description

This audio bug is hard to spot. When multiple AudioStreamPlayers play different AudioStreams, the pitch or volume of certain frequencies of the AudioStream get altered. 
I've provided a video below to show the issue better.

https://github.com/user-attachments/assets/11866800-dc11-4011-97f2-3ee54e52349f

In the video, a brownian noise is played constantly, with no major change in pitch or volume. When a pluck sound effect is played, however, you can hear the pitch of the brownian noise get very slightly altered. I suspect this happens to the audio frequencies shared by both sound effects, as it is easier to spot this bug if two AudioStreams have a very similar pitch range.

### Steps to reproduce

1. Start with a scene of your liking
2. Add two AudioStreamPlayers
3. Add two AudioStreams to the Players. (I recommend using a constant audio sample and a short sound effect, like a pluck)
4. Play the AudioStreamPlayers, and make sure both AudioStreams are audible at the same time
5. Try to spot the alteration in audio/volume

### Minimal reproduction project (MRP)

[audio-pitch-issue.zip](https://github.com/user-attachments/files/16789278/audio-pitch-issue.zip)

This is the same project I've used in the video. To replicate the issue, simply set the "playing" property of the first AudioStreamPlayer to True, and then do the same to the second one. The bug happens both in editor and in the running application.

**Comment by nack098** [comment]
It's really hard to see the different but what I heard was the pitch did went down after the pluck right?
Please correct me if I wrong.

**Comment by Kadmioo** [comment]
Yes, it seems that the pitch of the noise goes down, or at the very least its timbre changes. Sorry if I'm responding so late, I've been very busy with work on other projects.

**Comment by Kadmioo** [comment]
I've been doing some research, and I would like to make the hypothesis that the cause is [phase cancellation](https://en.wikipedia.org/wiki/Wave_interference). I tested the two sound effects in Audacity too, and noticed the issue is also present. BUT, the difference is that, in Audacity, the brownian noise shortly goes back to its original timbre, while in Godot the timbre of the noise is permanently modified unless the audio stream is stopped and restarted.

**Comment by Grublady** [comment]
Unable to reproduce with the MRP on 4.3 or 4.5-dev2, macOS 15.4.

Looks like some kind of compression or auto-gain is being applied somewhere in the system's audio chain—
This is likely a Windows-specific issue.

(The problem seems unrelated to phase cancellation.)

CC @AThousandShips

**Comment by nack098** [comment]
It is highly possible it might happen from the Windows audio system.
For the phase cancellation. it should be happen only at that moment when that sound play, in other word, the effect should not persist.
In reality phase cancellation in this case is already hard if the source itself didn't misaligned to some very prominent degree.

At first, I thought something might making the signal to change it frequency but from above response that wouldn't be the case.

I think we need more information on this but currently I will try to point my finger to the windows os api implementation first.

**Comment by Kadmioo** [comment]
Can confirm I cannot recreate the issue on Linux either.

**Comment by Akosmo** [comment]
Can't reproduce this with the MRP
Godot v4.7.rc2 - Windows 10 (build 19045) - WASAPI (48000 Hz, Stereo/mono)
